### DISCUSSION TITLE
可否配置 `.sumi` 或者 `extensions` 的磁盘位置？

### DISCUSSION THREAD

**Discussion by fankangsong** [opensumi/core#2568] [Q&A] [answered]
基于 https://github.com/opensumi/ide-startup 项目，在 Windows 环境 build 完运行 `node dist-node/server/index.js`，访问页面时 node 的错误日志：

```
[error] [Error: ENOENT: no such file or directory, scandir 'C:\Users\fanks\AppData\Roaming\.sumi\extensions'] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'C:\\Users\\fanks\\AppData\\Roaming\\.sumi\\extensions'
}
```

手动把工程里下载的 extensions 目录拷贝到 .sumi 目录后，插件、图标主题以及代码高亮才可正常工作。是否可以配置 .sumi 或者 extensions 的路径？

**Comment by PerfectPan** [comment, marked as answer]
https://github.com/opensumi/ide-startup/blob/main/src/browser/render-app.ts#L23 renderApp 应该可以配置 extensionDir, storageDirName, extensionStorageDirName 来完成自定义目录存储

可以看下 TS 定义：https://github.com/opensumi/core/blob/main/packages/core-browser/src/react-providers/config-provider.tsx#L53

**Comment by erha19** [comment]
@fankangsong 也可以看一下这个文档，是支持配置插件目录(extensionDir)的，https://opensumi.com/zh/docs/integrate/universal-integrate-case/custom-config#%E9%9B%86%E6%88%90%E9%85%8D%E7%BD%AE

**Comment by fankangsong** [comment]
marketplace.extensionDir 可以定义扩展目录，谢谢。
